src/main/java/org/jabref/logic/citation/repository/MVStoreBibEntryRelationDAO.java

```diff
@@ -0,0 +1,238 @@
+            .orElse(List.of());
+    }
+
+    /**
+     * Allows insertion of empty list in order to keep track of insertion date for an entry.
+     *
+     * @param entry should not be null
+     * @param relations should not be null
+     */
+    @Override
+    synchronized public void cacheOrMergeRelations(BibEntry entry, List<BibEntry> relations) {
+        entry.getDOI().ifPresent(doi -> {
+            try (var store = this.storeConfiguration.open()) {
+                // Save the relations
+                MVMap<String, LinkedHashSet<BibEntry>> relationsMap = store.openMap(mapName, mapConfiguration);
+                var relationsAlreadyStored = relationsMap.getOrDefault(doi.getDOI(), new LinkedHashSet<>());
+                relationsAlreadyStored.addAll(relations);
+                relationsMap.put(doi.getDOI(), relationsAlreadyStored);
+
+                // Save insertion timestamp
+                var insertionTime = LocalDateTime.now(TIME_STAMP_ZONE_ID);
+                MVMap<String, LocalDateTime> insertionTimeStampMap = store.openMap(insertionTimeStampMapName);
+                insertionTimeStampMap.put(doi.getDOI(), insertionTime);
+
+                // Commit
+                store.commit();
+            }
+        });
+    }
+
+    @Override
+    synchronized public boolean containsKey(BibEntry entry) {
+        return entry
+            .getDOI()
+            .map(doi -> {
+                try (var store = this.storeConfiguration.open()) {
+                    MVMap<String, LinkedHashSet<BibEntry>> relationsMap = store.openMap(mapName, mapConfiguration);
+                    return relationsMap.containsKey(doi.getDOI());
+                }
+            })
+            .orElse(false);
+    }
+
+    @Override
+    synchronized public boolean isUpdatable(BibEntry entry) {
+        var clock = Clock.system(TIME_STAMP_ZONE_ID);
+        return this.isUpdatable(entry, clock);
+    }
+
+    @VisibleForTesting
+    boolean isUpdatable(final BibEntry entry, final Clock clock) {
+        final var executionTime = LocalDateTime.now(clock);
+        return entry
+            .getDOI()
+            .map(doi -> {
+                try (var store = this.storeConfiguration.open()) {
+                    MVMap<String, LocalDateTime> insertionTimeStampMap = store.openMap(insertionTimeStampMapName);
+                    return insertionTimeStampMap.getOrDefault(doi.getDOI(), executionTime);
+                }
+            })
+            .map(lastExecutionTime ->
+                lastExecutionTime.equals(executionTime) || lastExecutionTime.isBefore(executionTime.minusWeeks(1))
+            )
+            .orElse(true);
+    }
+
+    private static class BibEntrySerializer extends BasicDataType<BibEntry> {
+
+        private final static String FIELD_SEPARATOR = "--";
+
+        private static String toString(BibEntry entry) {
+            return String.join(
+                FIELD_SEPARATOR,
+                entry.getTitle().orElse("null"),
```

Comment:
Mhm.. What if a title of a paper is `null`? Or author.

In practice -- of course not, but still there are such cases